test/gt2py_fo_exec.jl

```diff
@@ -19,73 +20,147 @@ macro to_py(expr::Expr)
+# Arguments
+- `setupfunc::Union{Function,Nothing}`: An optional setup function. If provided, it will be called before the test function. 
+                                        If `nothing`, the test function is called directly.
+- `testfunc::Function`: The test function to be executed.
+- `args...`: Additional arguments to be passed to the test function.
 
-offset_provider = Dict{String, Connectivity}(
-                   "E2C" => E2C_offset_provider,
-                   "C2E" => C2E_offset_provider
-                )
+# Usage
+- If `setupfunc` is provided, it should return the necessary data that `testfunc` will use. 
+  The returned data will be passed as the first argument to `testfunc`, followed by `args...`.
+- If `setupfunc` is `nothing`, `testfunc` will be called directly with `args...`.
 
+# Examples
 
-# Tests ------------------------------------------------
-function test_gt4py_fo_exec()
-    a = Field(Cell, collect(1.:15.))
-    b = Field(Cell, collect(-1.:-1:-15.))
+## Example 1: Without Setup Function
+```julia
+function mytest(args...)
+    println("Running test with arguments: ", args)
+end
+
+testwrapper(nothing, mytest, 1, 2, 3)
+# Output: Running test with arguments: (1, 2, 3)
+```
+
+## Example 2: With Setup Function
+```julia
+function setup()
+    return "setup data"
+end
+
+function mytest(data, args...)
+    println("Setup data: ", data)
+    println("Running test with arguments: ", args)
+end
+
+testwrapper(setup, mytest, 1, 2, 3)
+# Output: 
+# Setup data: setup data
+# Running test with arguments: (1, 2, 3)
+```
+"""
+function testwrapper(setupfunc::Union{Function,Nothing}, testfunc::Function, args...)
+    if setupfunc === nothing
+        testfunc(args...)
+    else
+        data = setupfunc()
+        testfunc(data, args...)
+    end
+end
+
+# ========================================
+# ============== Setup ===================
+# ========================================
+
+function setup_simple_connectivity()
+    edge_to_cell_table = [
+        [1 -1];
+        [3 -1];
+        [3 -1];
+        [4 -1];
+        [5 -1];
+        [6 -1];
+        [1 6];
+        [1 2];
+        [2 3];
+        [2 4];
+        [4 5];
+        [5 6]
+    ]
+
+    cell_to_edge_table = [
+        [1 7 8];
+        [8 9 10];
+        [2 3 9];
+        [4 10 11];
+        [5 11 12];
+        [6 7 12]
+    ]
+
+    E2C_offset_provider = Connectivity(edge_to_cell_table, Cell, Edge, 2)
+    C2E_offset_provider = Connectivity(cell_to_edge_table, Edge, Cell, 3)
+
+    offset_provider = Dict{String,Connectivity}(
+        "E2C" => E2C_offset_provider,
+        "C2E" => C2E_offset_provider,
+        "E2CDim" => E2C_offset_provider #TODO(lorenzovarese) this is required for the embedded backend (note: python already uses E2C)
+    )
+
+    return ConnectivityData(edge_to_cell_table, cell_to_edge_table, E2C_offset_provider, C2E_offset_provider, offset_provider)
+end
+
+# ========================================
```

Comment:
Please use a coherent format everywhere, the rest of the code base is in this style: https://github.com/GridTools/GridTools.jl/blob/main/src/GridTools.jl#L39
Pick the one you like.